src/moments.jl

```diff
@@ -363,3 +363,40 @@ end
 
 kurtosis(v::RealArray) = kurtosis(v, mean(v))
 kurtosis(v::RealArray, wv::AbstractWeights) = kurtosis(v, wv, mean(v, wv))
+
+"""
+    cumulant(v, k, [wv::AbstractWeights], m=mean(v))
+
+Return the `k`th order cumulant of a real-valued array `v`, optionally
+specifying a weighting vector `wv` and a pre-computed mean `m`. If `k` is a `UnitRange`, then return all the cumulants of orders in this range as a vector.
+
+This quantity is calculated using a recursive definition on lower-order cumulants and central moments.
+"""
+function cumulant(v::RealArray, krange::UnitRange{Int}, wv::AbstractWeights, m::Real=mean(v, wv))
+    if minimum(krange) <= 0
+        throw(ArgumentError("The cumulant order must be positive."))
+    end
+    k = maximum(krange)
+    cmoms = zeros(typeof(m), k)
+    cumls = zeros(typeof(m), k)
+    cmoms[1] = 0
+    cumls[1] = m
+    for i = 2:k
+        if wv isa UnitWeights
+            kn =  moment(v, i, m)
+        else
+            kn =  moment(v, i, wv, m)
+        end
+        cmoms[i] = kn
+        for j = 2:i-2
+            kn -= binomial(i-1, j)*cmoms[j]*cumls[i-j]
+        end
+        cumls[i] = kn
+    end
+    return cumls[krange]
+end
+
+cumulant(v::RealArray, krange::UnitRange{Int}, m::Real=mean(v)) = cumulant(v, krange, uweights(typeof(m), length(v)), m)
+
+cumulant(v::RealArray, k::Int, wv::AbstractWeights, m::Real=mean(v, wv)) = first(cumulant(v, k:k, wv, m))
+cumulant(v::RealArray, k::Int, m::Real=mean(v)) = cumulant(v, k, uweights(typeof(m), length(v)), m)
```

Comment:
I think you can simply drop these definitions and instead use a wider signature above: `krange::Union{Integer, AbstractRange{<:Integer}}`. This should work as numbers behave as single-element collections.